### PR TITLE
Added replacable resource locator strategy

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | yes
| Fixed tickets | none
| Related issues/PRs | based on https://github.com/sulu/sulu/pull/2729
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR add the ability to replace resource-locator strategy for each webspace (configurable over xml).

#### Example Usage
 
```php
$strategyManager = $container->get('sulu.content.rlp.strategy_manager');
$strategy = $strategyManager->getStrategyByWebspaceKey('sulu_io');
```

#### BC Breaks/Deprecations

Avoid using the strategy directly. Use the manager instead and get the strategy from there.

### CODE REVIEW

**Comment by wachterjohannes** [comment]
@danrot i also have changed the tree strategy name in this PR

**Comment by danrot** [comment]
I only have two naming things to complain about:

* I would not introduce another `Manager`, since we already have far too much of them. The `StrategyManager` simply holds some services and returns them, so I would call it a `ResouceLocatorStrategyRegistry`.
* I think we already agreed on renaming `Rlp` to `ResourceLocator`, so I would not use `rlp` in all the variable names and so on, but immediately replace them with `resourceLocator`.

**Comment by danrot** [comment]
There are still quite a lot variables called `rlp...`. Also, you named a lot variables simply `$strategy`, I would call them `$resourceLocatorStrategy`, to make sure they are not confused with another strategy.

**Comment by wachterjohannes** [comment]
@danrot i have renamed all the `$rlp*` hopefully the tests go green (:

**Comment by wachterjohannes** [comment]
@danrot ping

**Comment by danrot** [comment]
There are still some `$strategy` variables around. And you haven't renamed all the `ResourceLocatorStrategyManager` variables.

File: src/Sulu/Component/Content/Types/ResourceLocator/Strategy/StrategyNotExistsException.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Types\ResourceLocator\Strategy;
+
+/**
+ * Will be raised if the requested strategy is not available.
+ */
+class StrategyNotExistsException extends \Exception
```
Comment (danrot) [inline src/Sulu/Component/Content/Types/ResourceLocator/Strategy/StrategyNotExistsException.php]: WDYT about `ResourceLocatorStrategyNotFoundException`?

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**